logs_analyzer/app/scala/src/main/scala/com/databricks/apps/logs/ApacheAccessLog.scala

```diff
@@ -0,0 +1,42 @@
+package com.databricks.apps.logs
+
+/**
+  * An entry of Apache access log.
+  */
+
+case class ApacheAccessLog(ipAddress: String,
+                           clientIdentd: String,
+                           userId: String,
+                           dateTime: String,
+                           method: String,
+                           endpoint: String,
+                           protocol: String,
+                           responseCode: Int,
+                           contentSize: Long) extends Serializable {
+}
+
+object ApacheAccessLog {
+  private val PATTERN = """^(\S+) (\S+) (\S+) \[([\w:/]+\s[+\-]\d{4})\] "(\S+) (\S+) (\S+)" (\d{3}) (\d+)""".r
+
```

Comment:
I took Scala sources as a reference. I believe, Scala authors inherited their styles from Sun code style guidelines for Java, specifically: http://www.oracle.com/technetwork/java/javase/documentation/index-137868.html#format.